api/v1/porterconfig_types.go

```diff
@@ -91,12 +91,12 @@ func MergeMap(target, override map[string]interface{}) map[string]interface{} {
 
 // SecretsConfig is the plugin stanza for secrets.
 type SecretsConfig struct {
-	PluginConfig `json:",squash" yaml:",inline" mapstructure:",squash"`
+	PluginConfig `json:",inline" yaml:",inline" mapstructure:",squash"`
```

Comment:
This is always confusing! 😅  

I am pretty sure that the json tag does not support squash or inline. Embedded fields are always inlined and do not require a json entry at all.

I'm pretty sure that the json tag just ignores the `,inline` anyway but it may be more clear (so that people aren't tricked into thinking it is doing anything) if we remove that json tag entirely.

```suggestion
	PluginConfig `yaml:",inline" mapstructure:",squash"` # json is inline automatically because it's an embedded field
```

Reply:
I think you are right on the embedded fields are always inlined and do not require a json entry when used with `encoding/json`, the standard library.
However, it looks like the `controller-gen` requires it to be specified. I found the same `json:,inline; in their doc as well. https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html

Reply:
Huh I wonder what they are doing that it's used? 🤔 

Thanks for checking, I'll try to remember this the next time I'm tempted to remove it!